### PR TITLE
fix: solve the problem of push external repository fail

### PR DESCRIPTION
**Pull request by littlejiancc**
#### What type of this PR
/kind bug

#### What this PR does / why we need it:
fix: solve the problem of push external repository fail

#### Which issue(s) this PR fixes:
https://erda-org.erda.cloud/erda/workBench/projects/387/issues/all?id=72245&issueFilter__urlQuery=eyJzdGF0ZUJlbG9uZ3MiOlsiT1BFTiIsIldPUktJTkciLCJXT05URklYIiwiUkVPUEVOIiwiUkVTT0xWRUQiXSwiYXNzaWduZWVJRHMiOlsiMTAwMTI2MSJdfQ%3D%3D&issueTable__urlQuery=eyJwYWdlTm8iOjF9&issueViewGroup__urlQuery=eyJ2YWx1ZSI6ImthbmJhbiIsImNoaWxkcmVuVmFsdWUiOnsia2FuYmFuIjoiZGVhZGxpbmUifX0%3D&iterationID=429&type=BUG

- Fixes #your-issue_number
- [Erda Cloud Issue Link](paste your link here)

#### Specified Reviewers:

/assign @sfwn 

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: modules/gittar/auth/auth.go [hunk 1]
```diff
@@ -242,7 +243,8 @@ func doAuth(c *webcontext.Context, repo *models.Repo, repoName string) {
 	//如果是内置账户不做校验
 	innerUser, err := GetInnerUser(c)
 	if err == nil {
-		gitRepository, err = openRepository(repo)
+		repoLock := models.RepoLock{Repo: repo, RwMutex: &sync.RWMutex{}}
```
Comment (sfwn) [inline modules/gittar/auth/auth.go:246]: Put mutex into repo struct is better.

**Review by sfwn — CHANGES_REQUESTED** [review]

File: modules/gittar/auth/auth.go [hunk 2]
```diff
@@ -197,7 +197,7 @@ func doAuth(c *webcontext.Context, repo *models.Repo, repoName string) {
 	for _, skipUrl := range conf.SkipAuthUrls() {
 		if skipUrl != "" && strings.HasSuffix(host, skipUrl) {
 			logrus.Debugf("skip authenticate host: %s", host)
-
+			repo.RwMutex = &sync.RWMutex{}
```
Comment (sfwn) [inline modules/gittar/auth/auth.go:200]: set rwmutex inside openRepository method.

**Review by sfwn — CHANGES_REQUESTED** [review]

File: modules/gittar/auth/auth.go (hunk 1 above)
Comment (littlejiancc) [reply modules/gittar/auth/auth.go]: done

File: modules/gittar/auth/auth.go (hunk 2 above)
Comment (littlejiancc) [reply modules/gittar/auth/auth.go]: done

**Review by sfwn — APPROVED** [review]

**Pull request merged**